### PR TITLE
Add harfbuzz shaping regression tests: locl for Cyrillic langs

### PR DESCRIPTION
**Pull request by chrissimpkins**
This PR adds harfbuzz regression testing definition files for the `locl` feature support of the following localized Cyrillic script forms:

- Bulgarian
- Bashkir
- Chuvash
- Macedonian
- Serbian

The data have not been reviewed in detail to confirm that bugs do not exist in the production fonts. This intent is to add the production font data for regression testing and future bug hunt support.

### Issues

~~The Bulgarian output is not appropriate.  It shows all of the default Cyrillic glyphs despite setting the script/lang appropriately.  I can verify that the production binaries have the correct localized behavior in Font Goggles when the `cyrl` script + `BGR` lang is selected.  Will need to troubleshoot this.~~ (fixed in https://github.com/googlefonts/googlesans/pull/164/commits/d6a692a8bfdf539cf3ff017bc1561d2f4484f2a7)

### Notes

- localized smcp forms will need to be added in a separate PR for the smcp feature (added in https://github.com/googlefonts/googlesans/pull/164/commits/2957e5528e5ef5c295f2da836eccad241282baa9)
- @madig is working on the Latin `locl` regression tests.  These can be pushed to this branch or added in a separate PR

Tracking: https://github.com/googlefonts/googlesans/issues/161

### CODE REVIEW

**Comment by madig** [comment]
Need to use BCP 47 language tags: https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry. Using "bg" instead of "BGR" will change the output for Bulgarian.

**Comment by madig** [comment]
@googlebot I consent.

**Comment by madig** [comment]
smcp and c2sc seem to be completely broken for BSH, CHU and SRB?!

**Comment by chrissimpkins** [comment]
> smcp and c2sc seem to be completely broken for BSH, CHU and SRB?!

😭

File: qa/shaping/locl_cyrl_BGR.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
       "locl": true
     },
     "script": "cyrl",
-    "language": "BGR",
+    "language": "bg",
```
Comment (chrissimpkins) [inline qa/shaping/locl_cyrl_BGR.json:10]: ❤️

**Comment by moyogo** [comment]
> Need to use BCP 47 language tags: https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry. Using "bg" instead of "BGR" will change the output for Bulgarian.

Harfbuzz can take special language tags like `x-hbotBGR` with `x-hbot` + OT language system tag.

**Comment by chrissimpkins** [comment]
> > Need to use BCP 47 language tags: https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry. Using "bg" instead of "BGR" will change the output for Bulgarian.
> 
> Harfbuzz can take special language tags like `x-hbotBGR` with `x-hbot` + OT language system tag.

ty!

**Comment by chrissimpkins** [comment]
> > Need to use BCP 47 language tags: https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry. Using "bg" instead of "BGR" will change the output for Bulgarian.
> 
> Harfbuzz can take special language tags like `x-hbotBGR` with `x-hbot` + OT language system tag.

Does case matter Denis?

**Comment by moyogo** [comment]
> Does case matter Denis?

No, "x-hbotBGR" and "x-hbotbgr" produce the same output.

There is an open issue to document these tags: https://github.com/harfbuzz/harfbuzz/issues/2680

**Comment by chrissimpkins** [comment]
Thanks Denis. There is some good info in that thread.

> You need to pass the sub-tag to language not script, e.g. --script=deva --language=x-hbscdev2



> > Is it mandatory to specify both --script= and --language=

> That is my understanding. Script is needed to select the proper shaper (if you don’t specify it, hb-view/hb-shape will guess it), and the private language subtag allows selecting specific OT script and/or language, bypassing the automatic mapping that harfbuzz does.

**Comment by alexeiva** [comment]
@madig Here is a list of glyphs that have locl subs by language. 

Both Roman and Italic:
* Bulgarian
`ДЛФвгджзийѝклфптцшщьъюы/uni0434.sc/uni043B.sc/uni0444.sc`

* Bashkir
`ҒҘҪғҙҫ/uni0493.sc/uni0499.sc/uni04AB.sc`

* Chuvash
`Ҫҫ/uni04AB.sc`

Roman Forms:
* Macedonian
none
* Serbian
`б`

Italic Forms:
* Macedonian
`гѓбдптш`
* Serbian
`бдптш`

**Comment by alexeiva** [comment]
Here I've combined the list for all languages:
```
ДЛФҒҘҪ
вгджзийѝклфптцшщьъюығҙҫгѓбдптш
/uni0434.sc/uni043B.sc/uni0444.sc
/uni0493.sc/uni0499.sc/uni04AB.sc
```
Plus a separate list for Chuvash to avoid a conflict with Bashkir substitutions
`Ҫҫ/uni04AB.sc`

**Comment by madig** [comment]
Thanks! So we can boil the cyrl strings down to `ДЛФҒҘҪвгджзийѝклфптцшщьъюығҙҫгѓбдптш` and catch everything (for each language, smcp/c2sc on and off).

**Comment by alexeiva** [comment]
> Thanks! So we can boil the cyrl strings down to `ДЛФҒҘҪвгджзийѝклфптцшщьъюығҙҫгѓбдптш` and catch everything (for each language, smcp/c2sc on and off).

Yes. And you don't need a separate list for Chuvash as long as you test each language individually.

**Comment by chrissimpkins** [comment]
I think that we may have captured all of these localized forms in the regression tests that were added in this PR.  I'll go through the lists and confirm. Thanks Alexei.

**Comment by madig** [comment]
My smcp tests can likely be trimmed.

**Pull request merged**